apps/src/blockly/addons/cdoSerializationHelpers.js

```diff
@@ -25,13 +25,15 @@ export function convertXmlToJson(xml) {
 
   const stateToLoad = Blockly.serialization.workspaces.save(tempWorkspace);
 
-  // Create a map of ids (key) and block serializations (value).
-  const blockIdMap = stateToLoad.blocks.blocks.reduce(
-    (map, blockJson) => map.set(blockJson.id, blockJson),
-    new Map()
-  );
-
-  addPositionsToState(xmlBlocks, blockIdMap);
+  if (xmlBlocks.length && stateToLoad.blocks) {
```

Comment:
Just curious if there's ever a case when one condition is `true` and the other is `false`.